pkg/agent/containerd/config_test.go

```diff
@@ -0,0 +1,748 @@
+package containerd
+
+import (
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/k3s-io/k3s/pkg/agent/templates"
+	"github.com/k3s-io/k3s/pkg/daemons/config"
+	"github.com/k3s-io/k3s/pkg/spegel"
+	"github.com/rancher/wharfie/pkg/registries"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_UnitGetHostConfigs(t *testing.T) {
+	type args struct {
+		registryContent   string
+		noDefaultEndpoint bool
+		mirrorAddr        string
+	}
+	tests := []struct {
+		name string
+		args args
+		want HostConfigs
+	}{
+		{
+			name: "no registries",
+			want: HostConfigs{},
+		},
+		{
+			name: "registry with default endpoint",
+			args: args{
+				registryContent: `
+				  mirrors:
```

Comment:
I know the inline YAML looks funny in GH, but they are correct when viewed in an editor that uses golang's suggested 2-space-per-tab alignment.